lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -191,7 +191,7 @@ private function fetchFromCache($rawCacheKey, \ReflectionClass $class)
     {
         $cacheKey = $rawCacheKey . self::$CACHE_SALT;
         if (($data = $this->cache->fetch($cacheKey)) !== false) {
-            if (!$this->debug || $this->isCacheFresh($cacheKey, $class)) {
+            if (!$this->debug && $this->isCacheFresh($cacheKey, $class)) {
```

Comment:
The original code seems correct: If we aren't in debug mode, then the cache freshness doesn't need to be checked. If we are in debug mode, then check the cache freshness.